src/sql/base/browser/ui/infoButton/infoButton.ts

```diff
@@ -0,0 +1,183 @@
+	description: string,
+	buttonMaxHeight: number,
+	iconClass: string,
+	iconHeight: number,
+	iconWidth: number,
+	iconPath: IUserFriendlyIcon,
+	textTitle: string,
+	buttonMaxWidth: number,
+}
+
+export class InfoButton extends sqlButton {
+	private _container: HTMLElement;
+	private _main: HTMLElement;
+	private _iconContainer: HTMLElement;
+	private _iconElement: HTMLElement;
+	private _textContainer: HTMLElement;
+	private _pTitle: HTMLElement;
+	private _pDesc: HTMLElement;
+
+	private _description: string = '';
+	private _buttonMaxHeight: number = 0;
+	private _iconClass: string = '';
+	private _iconHeight: number = 0;
+	private _iconWidth: number = 0;
+	private _iconPath: IUserFriendlyIcon = { light: '', dark: '' };
+	private _textTitle: string = '';
+	private _buttonMaxWidth: number = 0;
+
+	constructor(container: HTMLElement, options?: IInfoButtonOptions) {
+		super(container, options);
+		this._container = container;
+
+		{ // Creates the elements
+			this._container.style.display = 'flex';
+			this._container.style.justifyContent = 'space-around';
+
+			this._main = document.createElement('div');
+			DOM.addClass(this._main, 'flexContainer');
+			this._main.style.cursor = 'pointer';
+			this._main.style.backgroundColor = '#FFFFFF';
+			this._main.style.borderRadius = '4px';
+			this._main.style.boxShadow = '0px 1px 4px rgba(0, 0, 0, 0.14)';
+			this._main.style.padding = '10px';
+
+			this._iconContainer = document.createElement('div');
+			this._iconContainer.style.alignItems = 'flex-start';
+			this._iconContainer.style.display = 'flex';
+			this._iconContainer.style.flexFlow = 'column';
+			this._iconContainer.style.paddingRight = '10px';
+
+			this._iconElement = document.createElement('div');
+			DOM.addClass(this._iconElement, 'icon');
+
+			this._textContainer = document.createElement('div');
+			this._textContainer.style.color = '#006ab1';
+			this._textContainer.style.display = 'flex';
+			this._textContainer.style.flexFlow = 'column';
+			this._textContainer.style.justifyContent = 'space-between';
+			this._textContainer.style.padding = '0 0 0 10px';
+			this._textContainer.style.margin = '0px';
+
+			this._pTitle = document.createElement('p');
+			this._pTitle.setAttribute('aria-hidden', 'false');
+			this._pTitle.style.fontSize = '14px';
+			this._pTitle.style.fontWeight = 'bold';
+			this._pTitle.style.lineHeight = '20px';
+			this._pTitle.style.margin = '0px';
+
+			this._pDesc = document.createElement('p');
+			this._pDesc.setAttribute('aria-hidden', 'false');
+			this._pDesc.style.fontSize = '12px';
+			this._pDesc.style.lineHeight = '16px';
+			this._pDesc.style.margin = '0px';
+
+			this._textContainer.appendChild(this._pTitle);
+			this._textContainer.appendChild(this._pDesc);
+
+			this._iconContainer.appendChild(this._iconElement);
+
+			this._main.appendChild(this._iconContainer);
+			this._main.appendChild(this._textContainer);
+			this.element.appendChild(this._main);
```

Comment:
I am assuming you've looked into the vsbutton implementation, 'this.element' is an '<a>' element, you adding this whole thing to 'this.element' as a child. right?

Reply:
Yes. vsButton creates the anchor tag markup. Thankfully, someone exposed get element from within, so I can request that element and append my custom infoButton markup to it.